### ISSUE TITLE
Train on my viedo dataset

### ISSUE THREAD

**Issue by HansonXia** [facebookresearch/supervision-by-registration#62]
I have two datasets, One is a landmark labeled dataset and the other is a no labeled video dataset. Can I use supervision-by-registration to train my own dataset?
I try each batch containing one landmark data, and two video data. Landmark data calculation landmark loss and two video data calculation LK loss. Should I detach one of the  about video data?
Thanks

**Comment by D-X-Y** [comment]
Yes, you can, but you also need to detect the face bounding box at first. Would you mind to explain more about "detach one of the about video data"? The batch size regarding how many image and video frame in a batch is important to the performance.

**Comment by HansonXia** [comment]
@D-X-Y One batch has one-third of the landmark data, and two-thirds is video data (half is t-1 frame, half is t-frame), do I need to detach the t-1 frame? You mentioned the ratio of data between landmark and video, what proportion is appropriate?

**Comment by D-X-Y** [comment]
I think you don't need to detach t-1 frame. The ratio depends on the dataset and the detector. You need to have a try (sorry for that).

**Comment by HansonXia** [comment]
@D-X-Y If I use the reg network, is the input coordinate of LK a normalized value between 0 and 1, or is it relative to the absolute coordinate value of the original image?

**Comment by D-X-Y** [comment]
It should be the absolute coordinates on the associated feature maps.

**Comment by HansonXia** [comment]
@D-X-Y Do I need to adjust the values of fb_thresh, eps and forward_max? During training, because of the judgment of these three values, lk_loss is None. P.S. The coordinate value I‘m using is the absolute coordinate value mapped back to the original image.

**Comment by D-X-Y** [comment]
You could tune the fb_thresh. The default eps should be fine. You could try to increase forward_max if the LK visualization results are poor.

**Comment by HansonXia** [comment]
@D-X-Y Hi，I read your code carefully and found that when calculating lk loss, the input is the absolute coordinates on the 256x256 image. How to ensure the stability of its mapping back to the original image？

**Comment by yujingxin** [comment]
@D-X-Y 
Sorry to bother. I follow the instruction in README and I have a problem when using my own video data. The LK loss in training is always 0. Could you help me figure it out? 
The part of log is like:
``
-->>[Train]: [epoch-004-050][000/204] Time 18.83 (18.83) Data 3.93 (3.93) Forward 13.41 (13.41) Loss 41.1127 (41.1127) [LK= 0.0000 ( 0.0000)]  : L1=13.8581 : L2=13.5656 : L3=13.6890 [Time Left: 01:03:41] Vis-PTS : 67 (67.8) Ava-PTS : 0.0 (0.0)
 -->>[Train]: [epoch-004-050][010/204] Time 5.05 (6.30) Data 0.00 (0.36) Forward 5.01 (5.78) Loss 42.6591 (43.6153) [LK= 0.0000 ( 0.0000)]  : L1=14.8585 : L2=13.7662 : L3=14.0343 [Time Left: 00:20:15] Vis-PTS : 63 (64.3) Ava-PTS : 0.0 (0.0)
 -->>[Train]: [epoch-004-050][020/204] Time 5.05 (5.72) Data 0.00 (0.19) Forward 5.01 (5.43) Loss 46.6956 (44.0576) [LK= 0.0000 ( 0.0000)]  : L1=15.6621 : L2=15.3455 : L3=15.6880 [Time Left: 00:17:27] Vis-PTS : 67 (65.1) Ava-PTS : 0.0 (0.0)
 -->>[Train]: [epoch-004-050][030/204] Time 5.06 (5.51) Data 0.00 (0.13) Forward 5.02 (5.30) Loss 44.0108 (44.0162) [LK= 0.0000 ( 0.0000)]  : L1=14.8467 : L2=14.5607 : L3=14.6033 [Time Left: 00:15:52] Vis-PTS : 67 (65.4) Ava-PTS : 0.0 (0.0)
 -->>[Train]: [epoch-004-050][040/204] Time 5.10 (5.41) Data 0.00 (0.10) Forward 5.06 (5.25) Loss 44.6381 (44.1874) [LK= 0.0000 ( 0.0000)]  : L1=15.0087 : L2=14.7006 : L3=14.9288 [Time Left: 00:14:41] Vis-PTS : 68 (65.5) Ava-PTS : 0.0 (0.0)
 -->>[Train]: [epoch-004-050][050/204] Time 5.12 (5.35) Data 0.00 (0.08) Forward 5.07 (5.21) Loss 43.3750 (44.0321) [LK= 0.0000 ( 0.0000)]  : L1=14.6629 : L2=14.1752 : L3=14.5369 [Time Left: 00:13:38] Vis-PTS : 67 (65.4) Ava-PTS : 0.0 (0.0)
``

**Comment by D-X-Y** [comment]
It is because that the all tracked points are regarded as unreliable one. You should probably tune the hyper-parameters for LK thresholds.

**Comment by maxwell1995cao** [comment]
> Yes, you can, but you also need to detect the face bounding box at first. Wou
i want to know how to get the face bound,and why my lkloss still zero.thanks

**Comment by maxwell1995cao** [comment]
i want to know how to get the face bound,thx

**Comment by maxwell1995cao** [comment]
sorry to bother,my list have the same number of picture and the same config with demo,an why my lk-loss still 0.i try to tune the config,but can not change the lk loss.could help me ?
